### PR TITLE
Improvement/structure

### PR DESCRIPTION
**Pull request by duhnnie**
- Improves component structure by moving basic structure to a new superclass.
- Maintains search box state between calls.
- Caches user searches.
- Simplifies some action creators.

### CODE REVIEW

File: src/actions/listUsers.actions.js [hunk 1]
```diff
@@ -1,45 +1,69 @@
...
+export function clear() {
   return {
-    type: SEARCH_USER,
-    searchWord
+    type: ACTIONS.CLEAR_USERS
   };
 }
 
-export function loadUsers(listUsers) {
+export function search(query) {
```
Comment (duhnnie) [inline src/actions/listUsers.actions.js]: Now all search for users (default users and specific search users) are handled in the same function.

File: src/services/resources/users.js [hunk 2]
```diff
@@ -2,7 +2,7 @@ import ServiceAccessor from '../ServiceAccesor';
 
 const serviceAccessor = ServiceAccessor.getServiceAccessor();
 
-const token = 'Replace your personal token here';
+const token = 'TOKEN';
```
Comment (duhnnie) [inline src/services/resources/users.js]: It was changed to a single word 'TOKEN' to allow users a quickly select/paste of their tokens (by double-clicking the word)

File: src/components/search/search.js [hunk 3]
```diff
@@ -8,14 +8,14 @@ import { setSearchWord } from '../../actions/listUsers.actions';
...
+    this.props.history.push('/');
   }
 
   handleKeyPress(e) {
-    if (e.keyCode === 13 && this.props.searchWord) {
-      this.props.onSearch(this.props.searchWord);
-      this.props.history.push({pathname: '/'});
+    let query = e.currentTarget.value.trim();
+
+    if (e.keyCode === 13 && query) {
```
Comment (veymarjoel) [inline src/components/search/search.js]: 13 key code should be managed like a constant

File: src/components/search/search.js [hunk 4]
```diff
@@ -48,11 +48,11 @@ Search.propTypes = {
 };
 
 const mapStateToProps = state => ({
-  searchWord: state.listUsers.searchWord
+  searchWord: state.listUsers.term
```
Comment (veymarjoel) [inline src/components/search/search.js]: it seems searchWord is a more significant variable name to manage

File: src/containers/Home.js [hunk 5]
```diff
@@ -1,46 +1,59 @@
...
 
 // Actions
-import { getAllUsers } from '../actions/listUsers.actions';
+import { search, clear } from '../actions/listUsers.actions';
 
 import { connect } from 'react-redux';
 import { withRouter } from 'react-router-dom';
 
-class Home extends Component {
+class Home extends Page {
```
Comment (veymarjoel) [inline src/containers/Home.js]: It's better to manage composition instead of inheritance

File: src/containers/Page.js [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+    return null;
+  }
+
+  render () {
+    // TODO: should we use this.props.child
+    return (
+      <React.Fragment>
+        <TopBar/>
+        <Container>
+          { this.getContent() }
```
Comment (veymarjoel) [inline src/containers/Page.js]: With composition it would be better manage with this.props.children

File: src/components/search/search.js (hunk 4 above)
Comment (duhnnie) [reply src/components/search/search.js]: I think **term** is shorter, simpler and also has significance.

File: src/containers/Home.js (hunk 5 above)
Comment (duhnnie) [reply src/containers/Home.js]: You're right, we can discuss with other team members too. But due the soon deadline, a different ticket will be created to improve this in the next sprint.

created ticket: https://trello.com/c/yLnHzuTq

File: src/containers/Page.js (hunk 6 above)
Comment (duhnnie) [reply src/containers/Page.js]: You're right, we can discuss with other team members too. But due the soon deadline, a different ticket will be created to improve this in the next sprint.

created ticket: https://trello.com/c/yLnHzuTq

**Comment by duhnnie** [comment]
Check the PR comments again please.

**Pull request merged**